parachain/primitives/beacon/src/lib.rs

```diff
@@ -268,6 +269,54 @@ pub struct HeaderUpdate<
 	pub block_root_branch_header_root: H256,
 }
 
+#[derive(
+	Default,
+	Encode,
+	Decode,
+	CloneNoBound,
+	PartialEqNoBound,
+	RuntimeDebugNoBound,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(
+	feature = "std",
+	serde(deny_unknown_fields, bound(serialize = ""), bound(deserialize = ""))
+)]
+#[scale_info(skip_type_params(
+	FeeRecipientSize,
+	LogsBloomSize,
+	ExtraDataSize,
+	DepositDataSize,
+	PublicKeySize,
+	SignatureSize,
+	ProofSize,
+	ProposerSlashingSize,
+	AttesterSlashingSize,
+	VoluntaryExitSize,
+	AttestationSize,
+	ValidatorCommitteeSize,
+	SyncCommitteeSize
+))]
+#[codec(mel_bound())]
+pub struct HeaderUpdateCapella<
```

Comment:
Instead of creating a new `HeaderUpdate*` type for each new execution header format, can't we add the execution header type as a generic parameter?

I think this reduce code duplication in other parts of this PR.